### PR TITLE
Add API methods

### PR DESCRIPTION
**Pull request by dzianisv**

### CODE REVIEW

**Comment by kelostrada** [comment]
Why would you add the `opts` if you don't use them? By the way, I'm down to merge this PR too... Is the maintainer of this repo active?

**Comment by dzianisv** [comment]
I use it in another project, where this module is dependency.

**Comment by kelostrada** [comment]
Can you tell me how exactly that matters? :) I'm not hating or anything, just wanted to try to learn something new perhaps!

File: lib/gold.ex [hunk 1]
```diff
@@ -12,18 +12,18 @@ defmodule Gold do
   @doc """
   Starts GenServer link with Gold server.
   """
-  def start_link(config), do: GenServer.start_link(__MODULE__, config)
+  def start_link(config, opts \\ []), do: GenServer.start_link(__MODULE__, config, opts)
```
Comment (hickscorp) [inline lib/gold.ex:15]: I'm all :+1: for this as well. Being able to pass options is a standard, eg, for `name: :ing` the process...

File: lib/gold.ex [hunk 2]
```diff
@@ -12,18 +12,18 @@ defmodule Gold do
...
 
   @doc """
   Returns server's total available balance.
   """
   def getbalance(pid) do
     case GenServer.call(pid, :getbalance) do
```
Comment (hickscorp) [inline lib/gold.ex:21]: All those are extremelly usefull to stop the linter from complaining, :+1: on my side as well.

**Comment by dzianisv** [comment]
Up

**Comment by kelostrada** [comment]
hey guys I can merge those, but I still don't have authorization for hex repo. I'll try to contact @solatis and see if he can give me those...

@denis4net can I ask you to correct the test to not use IO.puts?

**Pull request merged**